server/decoders/systems/eote.ts

```diff
@@ -0,0 +1,351 @@
+        strength: optional(forcePowerUpgradeDecoder),
+        magnitude: optional(forcePowerUpgradeDecoder),
+        duration: optional(forcePowerUpgradeDecoder),
+        range: optional(forcePowerUpgradeDecoder),
+        control: optional(array(object({ description: stringWithLength(1) }))),
+        mastery: optional(forcePowerUpgradeDecoder)
+    }), {}),
+    reference: referenceDecoder,
+    ...supplementalDecoderPartial
+});
+
+// ---------------------------------------------------------------------------------------------------------------------
+// Characters
+// ---------------------------------------------------------------------------------------------------------------------
+
+const baseSysDetailsPartial = {
+    career: stringWithLength(0, 255),
+    species: stringWithLength(0, 255),
+    characteristics: withDefault(object({
+        brawn: boundedInteger(0, 10),
+        agility: boundedInteger(0, 10),
+        intellect: boundedInteger(0, 10),
+        cunning: boundedInteger(0, 10),
+        willpower: boundedInteger(0, 10),
+        presence: boundedInteger(0, 10)
+    }), characteristics),
+    experience: withDefault(object({
+        total: positiveInteger,
+        available: positiveInteger
+    }), experience),
+    defenses: withDefault(object({
+        soak: positiveInteger,
+        melee: positiveInteger,
+        ranged: positiveInteger
+    }), defenses),
+    health: withDefault(object({
+        wounds: positiveInteger,
+        woundThreshold: positiveInteger,
+        strain: positiveInteger,
+        strainThreshold: positiveInteger,
+        criticalInjuries: array(eoteCriticalInjury),
+        stimsUsed: positiveInteger,
+        staggered: boolean,
+        immobilized: boolean,
+        disoriented: boolean
+    }), health),
+    abilities: withDefault(array(positiveInteger), abilities),
+    gear: withDefault(array(gearDecoder), gear),
+    armor: withDefault(object({
+        name: stringWithLength(0, 255),
+        description: optional(stringWithLength(1)),
+        defense: positiveInteger,
+        soak: positiveInteger,
+        hardpoints: positiveInteger,
+        encumbrance: positiveInteger,
+        rarity: positiveInteger,
+        attachments: withDefault(array(positiveInteger), []),
+        qualities: array(object({ id: positiveInteger, ranks: boundedInteger(1) })),
+        notes: optional(string)
+    }), armor),
+    weapons: withDefault(array(object({
+        name: stringWithLength(1, 255),
+        description: optional(stringWithLength(1)),
+        skill: stringWithLength(1, 255),
+        damage: positiveInteger,
+        criticalRating: positiveInteger,
+        range: eoteRangeDecoder,
+        encumbrance: positiveInteger,
+        rarity: positiveInteger,
+        attachments: withDefault(array(positiveInteger), []),
+        qualities: array(object({ id: positiveInteger, ranks: optional(boundedInteger(1)) })),
+        notes: optional(string)
+    })), weapons)
+};
```

Comment:
There's a lot of duplication in these decoders. I think it's possible to combine things like the 'weapons' in the baseSysDetailsDecoder and the other weapon decoder so the common fields are shared. It's a style thing that can be resolved later primarily.

Reply:
There are, but there's a reason for it. The reason being that splitting it up didn't work. The decoders simply didn't have the properties in them, despite TypeScript claiming everything was chill. So I need to investigate why that's happening; I think there's a transpiling issue going on, frankly.